homu/main.py

```diff
@@ -1239,13 +1233,23 @@ def queue_handler():
             return process_queue(states, repos, repo_cfgs, logger, buildbot_slots, db, git_cfg)
 
     os.environ['GIT_SSH'] = os.path.join(os.path.dirname(__file__), 'git_helper.py')
+    sshcfg = cfg_git.get('ssh_config_path')
+    if sshcfg is not None:
+        os.environ['HOMU_SSH_CONFIG'] = sshcfg
     os.environ['GIT_EDITOR'] = 'cat'
 
+    tmp_ssh_key = None
+    if git_cfg['local_git']:
+        tmp_ssh_key = tempfile.NamedTemporaryFile(prefix='homu-sshkey')
```

Comment:
It seems bad to not clean up a temporary file with key info, esp. since it's named. It would be nice to use a context manager to automatically close this file (if we have one i.e. if `git_cfg['local_git']`), and wrap the thread spawning code in a `with` block that uses said context manager.